src/Workspaces/Core/Portable/Workspace/Solution/Checksum_Factory.cs

```diff
@@ -14,13 +14,50 @@ namespace Microsoft.CodeAnalysis
     // all these are just helper methods
     internal partial class Checksum
     {
+        private static readonly Stack<IncrementalHash> s_hashStack = new Stack<IncrementalHash>();
```

Comment:
There's another way that objectpool leaks:

Object pool does not do synchronization.  So two people freeing at the same time can lead to one object being 'lost'.  It does this so that is can be fast, with the presumption that leaking data isnot costly.  

There is no way to return a value here stating what happened, because there's no way to know what happened (without at least the object-pool adding synchronization, which is not oing to happen) :)